docs/source/guide/tasks.md

```diff
@@ -102,6 +103,116 @@ http://example.com/text.txt
 
 {% enddetails %}
 
+
+## How to retrieve data
+
+There are several steps to retrieve the data to display in the `Object` tag. The data retrieval is also used in [dynamic choices](/templates/serp_ranking.html) and [labels](/templates/inventory_tracking.html). Use the following parameters in the `Object` tag.
+
+### `value` (required)
+
+The `value` parameter represents the source of the data. It can be plain text or a step of complex data retrieval system. It can be denoted using the following forms:
+`value` (required)
+
+#### Variables 
+
+In most cases, the `Object` tag has the value with one variable (prefixed with a $) in it. For example, `<Audio value="$audio" ... />`. All variables  point to field in task data. For example, `$audio => task.data.audio`
+
+!!! attention "important"
+    In most cases, the `Object` tag has the value with one variable (prefixed with a $) in it. For example, `<Audio value="$audio" ... />`. All the variables point to a field in task data. For example, `$audio => task.data.audio`.
+
+
+#### Text
+
+The value parameter can be a string. It is useful for `Header` and `Text`. 
+
+Also, you can use the content of the tag as value. It is useful for descriptive text tags and is applied for `Label` and `Choice`.
+
+For example:
+
+```html
+<Header>Label audio:</Header>
+<Header value="Label only fully visible cars" />
+<Text name="instruction" value="Label only fully visible cars" />
+<Label>cat</Label>
+
+<Choice>other</Choice>
+```
+
+#### Other cases
+
+1. The `value` parameter can be a text containing variables prefixed by $.
+
+    For example:
+    ```html
+    <Header value="url: $image"/>
+    ```
+
+2. The `value` parameter can also refer to nested data in arrays and dicts (`$texts[2]` and `$audio.url`). 
+
+    For example: 
+    ```html
+    <Image name="image" value="$images[0]"/>
+    ```
+
+3. The `value` parameter can include `Repeater` tag substitution, by default `{{idx}} ($audios[{{idx}}].url)`. 
+The `value` parameter can be a string. It is useful for `Header` and `Text`. 
+
+
+### `valueType` (optional)
+
+The `valueType` parameter defines how to treat the data retrieved from the previous steps.
+There are two options such as the  "url" and raw data. Currently the raw data input can be  "text” or "json”. The  “text” is used for `HyperText` and `Text` tags and "json" is used for `TimeSeries` tag. 
+
+For example:
+
+- Using “url”: `<Text name="text1" value="$text" valueType="url"/>` displays the text loaded by the URL.
+
+- Using “text”: `<Text name="text" value="$text" valueType="text"/>` displays the URL without loading the text.
```

Comment:
How about `valueType="csv|url"` and other options for `<Table>` tag? That's where we have started to sort out the changes

Reply:
It's complicated and unique. And I thought that we already added mention of Table params, but we didn't...
First we should improve `Table` docs page, because currently it's so empty =(